client/src/components/TabularViewComponent/TabularViewComponent.js

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import React, { Component } from "react";
+import { DetailsList, SelectionMode } from "@fluentui/react/lib/DetailsList";
+import LoaderComponent from "../LoaderComponent/LoaderComponent";
+
+import "./TabularViewComponent.scss";
+
+export class TabularViewComponent extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      isLoading: true,
+      items: this.generateItems(),
+      columns: this.generateColumns()
+    };
+  }
+
+  componentDidMount = () => {
+    this.setState({ isLoading: false });
+  }
+
+  generateItems = () => {
+    const items = [];
+    this.props.relationships.forEach(element => {
```

Comment:
Use `.map` rather than foreach.

Map will return you a list rather than you having to manually create one and push items into it.

Reply:
Done